Gems/Atom/Utils/Code/Source/ImageComparison.cpp

```diff
@@ -65,14 +75,7 @@ namespace AZ
 
             for (size_t i = 0; i < bufferA.size(); i += BytesPerPixel)
             {
-                // We use the max error from a single channel instead of accumulating the error from each channel.
-                // This normalizes differences so that for example black vs red has the same weight as black vs yellow.
-                const int16_t diffR = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[i]) - aznumeric_cast<int16_t>(bufferB[i])));
-                const int16_t diffG = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[i + 1]) - aznumeric_cast<int16_t>(bufferB[i + 1])));
-                const int16_t diffB = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[i + 2]) - aznumeric_cast<int16_t>(bufferB[i + 2])));
-                const int16_t maxDiff = AZ::GetMax(AZ::GetMax(diffR, diffG), diffB);
-
-                const float finalDiffNormalized = maxDiff / 255.0f;
+                const float finalDiffNormalized = CalcMaxChannelDifference(bufferA, bufferB, i) / 255.0f;
```

Comment:
nit: might want to cast the MaxChannelDiff up to a float.

Reply:
cast added!